### DISCUSSION TITLE
Mtex installation on M1 Mac

### DISCUSSION THREAD

**Discussion by asing123456** [mtex-toolbox/mtex#1285] [Ask Anything]
Dear All,

As a material researcher and beginner of Mtex (Matlab as well), I have been trying to install the Mtex on my MacBook air M1 machine (the only powerful machine I currently have). I have issues by using the common method and trying to follow the instruction on the webpage (https://mtex-toolbox.github.io/installation). Method 1 'library load disallowed by system policy', and Method 2 'Unable to compile mex files during startup_mtex' were not working so I am trying with Method 3 'Compiling the nfft' and stuck with the step 'Get nfft source from github'. It seems that the link is not valid any longer. May I ask if any one could help with this issue? Many thanks.

**Comment by kilir** [comment]
Hi,
yes, for arm-based computers you probably have to compile the nfft mex files yourself. However, I do not see why the github link to nfft would no work. Works for me.

Cheers,
Rüdiger

**Reply by skidder17** [reply]
I'm also having trouble installing on an M1 Mac. I'm running Matlab 2021b and OS Monterey. Has anyone successfully done this? I'm also trying to follow the directions above (" Compiling the nfft"), but I've had to go quite a bit off script from the directions given. At the moment I'm stuck on "Set the compilation parameters," getting the error "-bash: ./configure: No such file or directory". I'm pretty new to compiling, so I didn't even know things like, apparently, Xcode needs to be installed to do this.

**Reply by asing123456** [reply]
I am also stuck in this step as well. I am alsmost giving up and plan to get a second-hand PC...

**Reply by kilir** [reply]
This simply means that you are not in the right directory and you should 'cd' to the directory i.e. after cloning the git repository `cd nfft/`.

**Reply by skidder17** [reply]
Somehow I got it working, at least partially! I get an error message saying mex files aren't working, but at least some basic mtex functions are working (I was able to plot a map of orientations of quartz). I wish I could be clearer on how this was accomplished, but after struggling with the issues above I eventually gave up and just tried to load mtex again by downloading a precompiled version. The latest version didn't work, but 5.4.0 did... I'm not sure if any of the 100 things tried earlier (installing homebrew, Xcode, installing gcc, getting nfft source from github, etc.) were necessary for that to work. It is possible though to get it running (minus some mex functionality) on a M1 Macbook Air running OS Monterey and Matlab 2021b.

**Reply by asing123456** [reply]
Did you restart the Matlab and see if still working? I could get the Mtex installed for one go, while not working in the second time after restarting the Matlab...Matlab keeps on crashing...

**Reply by asing123456** [reply]
Somehow I got it working on 5.4.0 as well (at least some functions). 

The instruction for re-compiling nfft was written for someone with some understanding of 'coding' (not even sure if this is the right wording). However, for ppl like me who has very limited knowledge of 'coding', this instruction is still a bit too tricky to follow. I could really appreciate it if someone could help to add a bit more operation details to that instruction.

**Reply by kilir** [reply]
Hi,

I agree, that once we were writing these instructions, we assumed that users would at least be familiar with the very basic terminal usage and logics and be either able to do make the relevant changes according to their specific os version or report the relevant problem in the forum (e.g. error messages).

The key problem arises form the fact, that apple does not really offer a developer friendly environment when it comes to something outside the walled garden of the app store. Therefore, people developed package managers like homebrew, fink or macports to actually allow users to easily install things which are not available from apple and too tedious to install manually. All voluntarily and free. However, in some (most?) constallations, those solutions also assume, that users at least an infinitesial understanding of how the operation system works at the userland side (e.g. things like the PATH variable, basic terminal usage, ...).

Given that there are many versions of macOS around and given that we cannot cover all possible variations, I may try to update the relavant parts of the instruction such that might become clearer what people may have to change adapt. But it may require that the user may also use the google, since - at least from my side - I don't have the capacity to start from the big bang birth of the universe and I don't have an apple M1 to try those thigns out.

We would like Mtex to be as easy as possible to run on many different operation systems. We can cover those, which we have available and for which someone volunteered to compile nfft. Since as far I know, nobody here uses an apple m1 computer, and accodingly, pre-compiled binaries are not available yet.

Some Options:
1) find someone who can build those on your computer, test them and provide them to Mtex so we might be able to provide an AArch64 version

2) if someone has an M1 (with matlab, the potential to wipe the machine afterwards) but doesn't know how to build the nfft, to provide ssh access to this machine so someone able to build them can use do that remotely

3) someone send us an M1 (preferred option)


Cheers,
Rüdiger

**Reply by asing123456** [reply]
Dear Rüdiger,

It is really nice to receive your reply with a very detailed explanation. It will be really great that some clearer instruction could be made (if possible). We (as someone really have zero knowledge in the operation system) will then work from that (with the aid of google, certainly) to make it working. 

I am currently working on option 1. A friend of mine (I believe is more familiar with the OSX operation system) just agreed to help dig into this issue (with no guarantee of working though). Hopefully we could get some solution and could post it to the forum. Some clearer instruction could certainly help us a lot. 

Cheers,
Kai

**Reply by kilir** [reply]
is that already clearer?
https://github.com/mtex-toolbox/mtex-toolbox.github.io/blob/master/pages/download/installation.md

**Comment by zmichels** [comment]
I struggled with this, too. Building them myself never worked right.
I finally got the startup warning to go away (and everything seems to be working) by using files from this download (precompiled binaries for Matlab) to replace files in the MTEX folder. 

https://www-user.tu-chemnitz.de/~potts/nfft/download.php#matlab

**Reply by kilir** [reply]
Hi Zach,
as far as I can see, there's no macos specific aarch64 either, is there?

What was your reason trying to build nfft from source?

Cheers, Rüdiger

**Reply by asing123456** [reply]
This magic trick from Zach get Mtex 5.70 working! However, Matlab still crashes each time. I will follow Rüdiger's revised instruction once have the time and try to get it working properly (hopefully)...

**Comment by zmichels** [comment]
A while back (couple months?) I had crashing issues (an MTEX computation would crash Matlab)… and at the same time… every time I started up Matlab, MTEX would give a warning about couldn’t use NFFT with OpenMP. So I tried to sort out the problem. It *seemed* ultimately that the start-warning issue depended more on my NFFT than OpenMP, tho I fiddled with them both. Eventually, I tried these “Matlab binaries” and just decided to replace any same-named files in MTEX folder with these, out of desperation. The next startup… no warning… no crashes. I’m not dying this is the *best* way to do things. But I thought I’d chime in with what seemed key to getting me back on track. That said… I don’t have an M1 Mac… but I did have some trouble with the last few installs in this regard.

Z

**Reply by kilir** [reply]
Hi Zach,
so it's the openMP-enabled version of nfft that comes with mtex that caused the warning. On what operation system is that?
Cheers,
Rüdiger

**Comment by asing123456** [comment]
I am quite curious: why Mtex 5.5.0 and Mtex 5.4.0 work fine (so far) on the M1 Mac...

**Reply by kilir** [reply]
Are you sure everything is actually working or does it just start up without telling you that not everything is ok? 
```
mtexdata dubna
odf = pf.calcODF
```

**Reply by asing123456** [reply]
Everything is working just fine, so far. I have tried with different datasets from TSL and HKL systems and done texture calculation, KAM calcualtion and others. No issue occurred.

**Comment by zmichels** [comment]
For me, I first noticed this when Ininstalled mTEX 5.something, and I think was running Mac OS 11… I’m on OS 12 now. For a while I just ignored the warning because everything still worked fine. Then a few months ago, I had some routine that would cause Matlab to crash on occasion, and I think someone suggested it might be related to the warning message. I compiled the libraries and bits multiple times and tried to get them working with Matlab, but I would still get the warning from MTEX at startup. Swapping the files into the MTEX locations was the only thing that made it go away.

Z

**Comment by ThWang77** [comment]
Hi,
I have an M1 macbook and an Intel macbook borrowed from my colleague. Both of their OS are 12.0.1 Monterey using Matlab2021b and either of them have compiled any source code. I modified some code and Intel macbook compiled nfft successuflly. Here is my procedure.
Following the procedure in https://mtex-toolbox.github.io/installation and installed gcc, fftw, autoconf, gcc libtool, automake using homebrew
` brew install gcc`
 `brew install fftw`
 `brew install gcc libtool`
 `brew install autoconf`
 `brew install automake`
Fftw is a library needed for compiling nfft. Autoconf and automake are library used in compilation for an mac never compiled code before. Before running ./bootstrap.sh in terminal, open bootstrap.sh, change line 36 'libtoolize' to 'glibtoolize'. Then follow step2, run ./bootstrap.sh, ./bootstrap.sh will generate file configure used in step3. In step3 using
`CC=/usr/local/Cellar/gcc/11.2.0_3/bin/gcc-11`
`LDFLAGS=-Wl,-L/usr/local/Cellar/gcc/11.2.0_3/lib/gcc/11`
`./configure --with-matlab=/Applications/MATLAB_R2021b.app --enable-nfsoft\
 --enable-nfsft --enable-portable-binary --with-fftw3=/usr/local/Cellar/fftw/3.3.10 --with-matlab-fftw3-libdir=/usr/local/Cellar/fftw/3.3.10/lib --with-gcc-arch=x86_64`
When I compiled nfft at the first time, I got an error about not able to find fftw. I manully add the address of fftw, and it worked. Then following step4, 5, 6, make and link...

After all that, Mtex5.7 works well, and I tested with grain calculation and wave velocity calculation of my own data, also worked well, no crashing, no openMP warning.

There still something confused me. When both Xcode and gcc are installed, clang which is gcc in Xcode runs, not gcc you manully installed. CC and LDFLAGS ask to use gcc that I download through homebrew, but acctually it was clang compiling nfft. I don't know if 
`CC=/usr/local/Cellar/gcc/11.2.0_3/bin/gcc-11` 
`LDFLAGS=-Wl,-L/usr/local/Cellar/gcc/11.2.0_3/lib/gcc/11` 
works or not. And I didn't use openMP because Xcode disabled openMP. And I don't know how to enable it...


For M1 macbook, things get complicated. I did the same thing before step3 and all went well. At step3, I used the following command:
`CC=/opt/homebrew/Cellar/gcc/11.2.0_3/bin/gcc-11`
`LDFLAGS=-Wl,-L/opt/homebrew/Cellar/gcc/11.2.0_3/lib/gcc/11`
`./configure --with-matlab=/Applications/MATLAB_R2021b.app --enable-nfsoft\
  --enable-nfsft  --enable-portable-binary --with-fftw3=/opt/homebrew/Cellar/fftw/3.3.10 --with-matlab-fftw3-libdir=/opt/homebrew/Cellar/fftw/3.3.10/lib --with-gcc-arch=arm64`
Gcc and fftw have native arm64 edition now, so Homebrew wouldn't install x86_64 edition. An error appared when I used make command. The error log is

`ld: warning: ignoring file /Applications/MATLAB_R2021b.app/bin/maci64/libmx.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64`
`ld: warning: ignoring file /Applications/MATLAB_R2021b.app/bin/maci64/libmex.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64`
`ld: warning: ignoring file /Applications/MATLAB_R2021b.app/bin/maci64/libmat.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64`
`Undefined symbols for architecture arm64:`
`  "_mexAtExit", referenced from:`
`      _mexFunction in libnfft_la-nfftmex.o`
`  "_mexErrMsgTxt", referenced from:`
`      _mexFunction in libnfft_la-nfftmex.o`
`      _nfft_mex_get_int in libmatlab.a(libmatlab_la-args.o)`
`      _nfft_mex_get_double in libmatlab.a(libmatlab_la-args.o)`
`      .............................`
`      .............................  %% A long list has been omitted...`
`  "_mxIsDouble", referenced from:`
`      _mexFunction in libnfft_la-nfftmex.o`
`      _nfft_mex_get_int in libmatlab.a(libmatlab_la-args.o)`
`      _nfft_mex_get_double in libmatlab.a(libmatlab_la-args.o)`
`      _nfft_mex_get_nm in libmatlab.a(libmatlab_la-args.o)`
`      _nfft_mex_get_nm_odd in libmatlab.a(libmatlab_la-args.o)`
`      _nfft_mex_get_n1n2m in libmatlab.a(libmatlab_la-args.o)`
`      _nfft_mex_get_n1n2m_odd in libmatlab.a(libmatlab_la-args.o)`
`      ...`
`"_mxIsScalar", referenced from:`
`_mexFunction in libnfft_la-nfftmex.o`
`"_mxIsSingle", referenced from:`
` _mexFunction in libnfft_la-nfftmex.o`
`"_mxMalloc", referenced from:`
`  _nfft_mex_malloc in libmatlab.a(libmatlab_la-malloc.o)`
`ld: symbol(s) not found for architecture arm64`
`clang: error: linker command failed with exit code 1 (use -v to see invocation)`
`make[2]: *** [libnfft.la] Error 1`
`make[1]: *** [all-recursive] Error 1`
`make: *** [all-recursive] Error 1`

I found one case compiling their source code for matlab encountered with same problem. 
[https://git.dynare.org/Dynare/dynare/-/issues/1780](url)
They thought that MATLAB didn't ready for ARM macbook caused the problem, so did I. As far as I thought, ignored files in first three rows **libmx.dylib, libmex.dylib, libmat.dylib** are x86_64 files designed for creating mex files that allow MATLAB to understand code written by other languages. When I checked error log, there were a lot of files (like **libnfft_la-nfftmex.o, libmatlab.a, libnfft_la-nfftmex.o**) that were already compiled files from nfft source code. Nfft can also be compiled in M1 Macbook, when you delete 
`--with-matlab=/Applications/MATLAB_R2021b.app` in step3. It's probably MATLAB's problem. I'm not an expert, don't have an idea how this could be fixed unless MATLAB releasing an native Apple silicon edition...

And unfortunately, Zach's replacing files method doesn't work for me, MATLAB still keeps crashing...

Tianheng

**Reply by kilir** [reply]
Hi Tianheng,

thank you for testing this all out. So it seems that this is rather a case for a Matlab issue.

The reason to use gcc from homebrew was exactly that it supports openMP while apple a) disabled openOP in clang shipped with xcode and b) does not supply libomp.dylib. On my old apple system back then, setting  CC and LDFLAGS seemed to have worked since the openMP version made use of all cores.

Cheers,
Rüdiger

Edit: since Matlab 2021b is running under rosetta2 (there's no native aarch64 Matlab yet), what sort of error does mtex give you for the mex binaries you compiled on the intel macbook if you use them on M1?

**Reply by ThWang77** [reply]
Hi Rüdiger,
I didn't know exactly which mex file was used during compilation. So I replaced whole files in Matlab, nfft, mtex5.7, using files I compiled in intel macbook. When using startup_mtex in matlab, there was only 'initialize MTEX 5.7.0  ....' in command window. No error message appeared but matlab crushed. When I restart matlab, Mtex initializing automatically and matlab keeps crushing. Same condition when I haven't replaced any files.

Tianheng

**Reply by kilir** [reply]
Hi,
ok, thanks for testing. So if I understand Matlab M1 works in general under rosetta2, however fatally fails using .mexmaci64. Maybe one needs to wait till Matlab supports the M1 natively.
Cheers,
Rüdiger

**Comment by asing123456** [comment]
Dear All, thanks so much for all kinds of supports to this Apple SIlicon compatibility issue. I guess the conclusion could be drawn: Either go back to older version (like 5.5.0), or sit tight waiting for the apple silicon native Matlab...

**Comment by kilir** [comment]
Hi,
included in mtex 5.8.1, there are some new binaries to be tries with an Apple arch chip. Could someone try this out?
Cheers,
Rüdiger

**Reply by asing123456** [reply]
I just tried with Matlab 2022a and this mtex 5.8.1, and confirmed that the basic functions like plotting are working fine so far.

**Reply by asing123456** [reply]
A quick update: after some simple analysis (2-3 times use), the Matlab back to crashing once starts...

**Reply by KaueSeo** [reply]
Hi!
I just got the M2 chip MacBook and using with either the 5.8.0 or 5.8.2 it does not compiled the nfft.. I can work a little, but when you come across some odf_ to do, it cannot plot or work. Does not crash, but cannot plot the CPO, for instance.

Anyone tried to solve this and had success?

al the best,
Kaue

**Comment by KaueSeo** [comment]
Hi!
I just got the M2 chip MacBook and using with either the 5.8.0 or 5.8.2 it does not compiled the nfft.. I can work a little, but when you come across some odf_ to do, it cannot plot or work. Does not crash, but cannot plot the CPO, for instance.

Anyone tried to solve this and had success?

al the best,
Kaue

**Comment by zmichels** [comment]
Yes, check the discussion forum for solutions people have found. For me it requires downloading other nfft files that have already been compiled for Matlab and swapping them into the MTEX folder. Then it works.On Feb 24, 2023, at 3:31 PM, KaueSeo ***@***.***> wrote:﻿
Hi!
I just got the M2 chip MacBook and using with either the 5.8.0 or 5.8.2 it does not compiled the nfft.. I can work a little, but when you come across some odf_ to do, it cannot plot or work. Does not crash, but cannot plot the CPO, for instance.
Anyone tried to solve this and had success?
al the best,
Kaue

**Reply by KaueSeo** [reply]
I see, I might have a very silly doubt, that can be related to my poor Matlab programming expertise, but I've downloaded the nfft files compiled from this site you've mentioned, but I can't see were it can be replaced in the MTEX folder. When I transfer them into the MTEX folder, nothing happens and the same problem keeps popping. 

By the way, thank you very much for your previous answer!

Cheers,
Kaue

**Comment by asing123456** [comment]
I am now using the Matlab apple silicon native beta version with Mtex 5.8.2 installed in both MacBook Pro M1 Max and MacBook Air M1, both working perfectly fine. 
The link for the apple silicon native beta version: https://www.mathworks.com/support/apple-silicon-r2022b-beta.html

**Reply by KaueSeo** [reply]
Hi, 
I did what you've said, downloaded the Matlab beta version, for silicon native operations, but still does not get the work done. The problem was in the same line as using the 'normal' Matlab.

Did you do something else, related to nfft, like compiling manually them or so?

Here are the errors:
Error in unimodalComponent/calcFourier>gcA2fourier (line 70)
plan = nfsoftmex('init',length(A)-1,length(g),nfsoft_flags,0,4,1000,2*ceil(1.5*(length(A)+1)));

Error in unimodalComponent/calcFourier (line 34)
f_hat = gcA2fourier(g,c,A);

Error in ODF/calcFourier (line 28)
  hat = calcFourier(odf.components{i},L,varargin{:});

Error in ODF/FourierODF (line 28)
f_hat = calcFourier(odf,varargin{:});

Error in orientation/calcFourierODF (line 54)
odf = FourierODF(odf,get_option(varargin,{'L','bandwidth','fourier'},L,'double'),varargin{:});

Error in orientation/calcDensity (line 71)
  odf = calcFourierODF(ori,varargin{:},'kernel',psi);

Error in KS176o2 (line 270)
odf_qz = calcDensity(ebsd_selected('Quartz').orientations,'halfwidth',10*degree);


Help is highly appreciated!
Thank you all!

Kaue

**Reply by asing123456** [reply]
Hi Kaue, to be honest, I am not sure how exactly I got Mtex working in the Matlab apple silicon native beta version. Probably some previous settings made it happen (I did really quite a lot). One thing to confirm, did you install Xcode? It seems essential to some of the relevant compiling.

**Reply by KaueSeo** [reply]
Hi, I understand. And yes I did install Xcode, which for the first steps in the macOS troubleshot page of MTEX it worked well, the problem arrives when I comes to the compiling itself. 

I was doing the manual swap of the similar archives of the MTEX folder with the NFFT compiled ones from the website put above, but still it didn't work. Maybe I forgot some archives to swap, I'll try it now again.

Thanks for answering!

**Reply by asing123456** [reply]
A silly question but want to confirm: have you tried to install the Matlab beta version and then Mtex 5.8.2, without any additional steps?

**Reply by KaueSeo** [reply]
Yes, this was actually the first step. Then it stuck in the same part as with the 'normal' Matlab, when the odf_ come to play, but I did have in the beginning the warning about the NFFT.

I mean, the Xcode was already in with the MEX correction, but no nfft modification.

**Reply by asing123456** [reply]
Okay...in that case, I may not have any further suggestions. Somehow it just works fine with my two machines...

**Reply by KaueSeo** [reply]
I see, that's ok though. I will try it again. 
Anyways, thanks for the effort.

**Comment by Ranjithvelm** [comment]
First of all, Thank you for all the suggestions. I have tried pretty much every suggestion.
I got an M2 Macbook air with Matlab_R2023a installed. 

I downloaded Mtex 5.9.0 using the following command
```zsh
curl -LJO --output mtex.zip https://github.com/mtex-toolbox/mtex/releases/download/mtex-5.9.0/mtex-5.9.0.zip
```
Extracted it.
Matlab crashes because of files in folders "mtex-5.9.0/extern/nfft" and "mtex-5.9.0/extern/nfft_openmp"
So I removed all the files in those two folders.

Then I downloaded the precompiled binaries from the official website in the terminal using the command
```zsh
curl -LJ0 --output nfft.zip https://www-user.tu-chemnitz.de/~potts/nfft/download/nfft-3.5.3-mexmaci64-openmp-core2.zip
```
Extracted it.
Then copied only the .mexmaci64 files from each folder inside "nfft-3.5.3-mexmaci64-openmp-core2/folder_with_the_mexmaci64 and pasted the files to mtex folder "mtex-5.9.0/extern/nfft_openmp"

Now, everything seems to work fine.

**Reply by B-Subhrajyoti** [reply]
Hello,
Does this above method works in the new [MATLAB R2023b Prerelease] which have native apple silicon support.
I tried the above method in my M1 MacBook air on 2023b prerelease version: 

When I started MTEX, it showed this following messages: 
>> startup_mtex
initialize MTEX 5.9.0  ....
********************************************************************************
 
  MTEX: Error running NFFT!
   
  The original error message was:
  Undefined function 'nfsftmex' for input arguments of type 'char'.
 
 I did not get NFFT working. This restricts the functionality of MTEX.
 To overcome this restriction you may need to compile the NFFT your own.
 Please have a look at https://mtex-toolbox.github.io/installation for possible workarounds

********************************************************************************

--------------------------------------------------------------------------------
Checking mex files failed!
> Trying now to recompile mex files.
... compiling S1Grid_find.c
... compiling S1Grid_find_region.c
... compiling S2Grid_find.c
... compiling S2Grid_find_region.c
... compiling SO3Grid_dist_region.c
... compiling SO3Grid_find.c
... compiling SO3Grid_find_region.c
... compiling insidepoly_dblengine.c
... compiling insidepoly_sglengine.c
... compiling adjoint_representationbased_coefficient_transform.c
... compiling representationbased_coefficient_transform.c

********************************************************************************
  MTEX: Couldn't get the mex files working!
   
  The original error message was:
  Undefined function 'SO3Grid_dist_region' for input arguments of type 'int32'.
 
 Please have a look at https://mtex-toolbox.github.io/installation
 or ask in the forum https://groups.google.com/forum/#!forum/mtexmail for help.

********************************************************************************

--------------------------------------------------------------------------------
 done!
 
MTEX 5.9.0 (show documentation)
  Import pole figure data
  Import EBSD data
  Import ODF data
 
  Install MTEX for future sessions

Does this mean it will work or not?

**Comment by Ranjithvelm** [comment]
When I tried to compile the NFFT binaries for Matlab_R2022b_Beta (apple silicon native beta version) using the command,
```
CC=/opt/homebrew/Cellar/gcc/12.2.0/bin/aarch64-apple-darwin22-gcc-12
LDFLAGS=-Wl,-L/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/12
./configure --with-matlab=/Applications/MATLAB_R2022b_Beta.app --enable-nfsoft
 --enable-nfsft  --enable-portable-binary --with-fftw3=/opt/homebrew/Cellar/fftw/3.3.10_1 
--with-matlab-fftw3-libdir=/opt/homebrew/Cellar/fftw/3.3.10_1/lib --with-gcc-arch=arm64
```
I got an error, 
```
unsupported file extension mexmaca64
```
Is there any way to overcome this error?

**Reply by asing123456** [reply]
You can try the 2023b pre-release with the apple silicon native version. It works well in my M2 MBA and M1MaxMBP. You need to install the Xcode as well.

**Reply by romydhan** [reply]
Hi, I've got Matlab R2023b installed, and Xcode, and trying to compile NFFT and am also stuck on this error ("unsupported file extension mexmaca64") - has anyone solved this yet?

**Reply by romydhan** [reply]
And, nevermind. Got around the issue by just downloading the precompiled binaries just posted a few days ago here:
https://github.com/mtex-toolbox/mtex/issues/1970

**Comment by lfgmorales** [comment]
Hi

Good that the trick on Matlab2023a works, will give it a try on my computer

But why then would you want to compile on 2022b, if you had a working version?

Or you are just testing it?

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 4 Apr 2023, at 06:31, Ranjithvelm ***@***.***> wrote:

﻿

When I tried to compile the NFFT binaries for Matlab_R2022b_Beta (apple silicon native beta version) using the command,

CC=/opt/homebrew/Cellar/gcc/12.2.0/bin/aarch64-apple-darwin22-gcc-12
LDFLAGS=-Wl,-L/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/12
./configure --with-matlab=/Applications/MATLAB_R2022b_Beta.app --enable-nfsoft
 --enable-nfsft  --enable-portable-binary --with-fftw3=/opt/homebrew/Cellar/fftw/3.3.10_1
--with-matlab-fftw3-libdir=/opt/homebrew/Cellar/fftw/3.3.10_1/lib --with-gcc-arch=arm64


I got an error,

unsupported file extension mexmaca64


Is there any way to overcome this error?

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1285#discussioncomment-5516700>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4CCZQMMFCOU73A6VDW7OP2TANCNFSM5IVUXOIA>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1285#discussioncomment-5516700", "url": "https://github.com/mtex-toolbox/mtex/discussions/1285#discussioncomment-5516700", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Ranjithvelm** [reply]
Hi
I was just testing it.
because the Rosetta emulation in Mac M2 takes up more memory to run Matlab.

So I tried to compile nfft for Apple silicon native beta version of matlab. 
here is the download link
```
https://www.mathworks.com/support/apple-silicon-r2022b-beta.html
```
